src/blockchain/TxStatus.ts

```diff
@@ -5,11 +5,17 @@
 /**
  * Dummy comment needed for correct doc display, do not remove
  */
+
+import { Codec } from '@polkadot/types/types'
+import ICType from '../types/CType'
+
+export type QueryResult = Codec | undefined | null | ICType | string
```

Comment:
After searching deep into the types from Polkadot, I looked into Codec and what it provides, then added some necessaries. Though the null and undefined may not be essential. Sparring on this topic would be ideal.

Reply:
```suggestion
export type QueryResult = Codec
```
Playing with chain queries it seemed to me that we only ever get Codec types back, even when the query is for something that is not on chain. In fact, it was going through some trouble in [this](https://github.com/KILTprotocol/sdk-js/pull/221) to make sure `null` is returned in these cases. Tests pass when QuerieResult is just `Codec`, including integrations tests.